src/main/java/frc/robot/RobotContainer.java

```diff
@@ -24,12 +32,28 @@
  * subsystems, commands, and trigger mappings) should be declared here.
  */
 public class RobotContainer {
+
   // The robot's subsystems are defined here...
   private final Drivetrain m_drive = new Drivetrain();
   private final FourBarArm m_arm = new FourBarArm();
 
   // Shuffleboard stuff
   SendableChooser<Command> m_autoCommand = new SendableChooser<>();
+  SendableChooser<Alliance> m_allianceChooser = new SendableChooser<>();
+
+  // Where the robot will score.
+  public static Node selectedNode = null;
+
+  // Array of april tags. The index of the april tag in the array is equal to its id, and aprilTags[0] is null.
+  public final static Pose3d[] aprilTags = new Pose3d[9];
```

Comment:
This code looks very suspect. It suggests that the Pose3d of tags 1-8 is (0, 0, 0) (0, 0, 0).

Furthermore, the array is not needed -- just call the Field method that provides Optional<Pose3d>.

Reply:
This array is filled later, after Vision.setup() is called. It just makes it easier to access an April tag. aprilTags[1] is a lot easier than Vision.getTagPose(1). It also avoids any possible error where reference variables aren't equal.